pkg/controller/dataimportcron-controller.go

```diff
@@ -471,6 +470,34 @@ func (r *DataImportCronReconciler) cleanup(ctx context.Context, dataImportCron *
 	return nil
 }
 
+func (r *DataImportCronReconciler) deleteJobs(ctx context.Context, dataImportCron *cdiv1.DataImportCron) error {
+	cronJobName := GetCronJobName(dataImportCron)
```

Comment:
Does this not work?
```go
cronJobName := GetCronJobName(dataImportCron)
deletePropagationBackground := metav1.DeletePropagationBackground
cronJob := &v1beta1.CronJob{ObjectMeta: metav1.ObjectMeta{Namespace: r.cdiNamespace, Name: cronJobName}}
if err := r.client.Delete(ctx, cronJob, &client.DeleteOptions{PropagationPolicy: &deletePropagationBackground}); IgnoreNotFound(err) != nil {
  return err
}
```

Reply:
Right, this will delete the current cronjob job, but not the initial job which is not a cronjob one. However, initial job should be deleted by name, so no need to list the jobs. Fixed. Thanks @awels .